lib/perl/Genome/SoftwareResult.pm

```diff
@@ -110,10 +110,28 @@ sub _faster_get {
             my $lookup_hash = $class->calculate_lookup_hash_from_arguments(@args);
             # NOTE we do this so that get can be noisy when called directly.
             @objects = $class->SUPER::get(lookup_hash => $lookup_hash);
+            if (@objects > 1) {
+                die $class->error_message(
+                    'Multiple results found for lookup hash %s!',
+                    $lookup_hash
+                );
+            }
+
+            if (@objects) {
```

Comment:
I'm not 100% sure I understand this suggestion, would you mind elaborating a bit?